### PR TITLE
Discard additional lines upon inserting when AcceptsReturn=false 

### LINKED ISSUE #5600: EditBox with AcceptReturn="false" accepts pasting text with CR(LF) characters.
**Describe the bug**
EditBox with AcceptReturn="false" accepts pasting text with CR-LF and transforms to multi-line EditBox.

**To Reproduce**
In any app select multi-line text.
Paste selection into EditBox

**Expected behavior**
CR(LF) should be removed when pasting text.

**Desktop (please complete the following information):**

- OS: [e.g. Windows, Mac, Linux (State distribution)]
- Windows

### PR DESCRIPTION
**Pull request by markl65536**
## What does the pull request do?
It changes text insertion into `TextBox` with `AcceptsReturn==false` so that behavior is the same as in WPF.

## What is the current behavior?
Upon insertion of multiline text into `TextBox` with `AcceptsReturn==false` the `TextBox` becomes multiline. It is currently impossible to prevent a `TextBox` becoming multiline by user input (e.g. by pasting from clipboard).

## What is the updated/expected behavior with this PR?
Upon insertion of multiline text into `TextBox` with `AcceptsReturn==false`, only the first line will be inserted and any additional lines will be discarded.

## How was the solution implemented (if it's not obvious)?
`TextBox.RemoveInvalidCharacters` was renamed to `SanitizeInputText` and extended to handle the multiline scenario.

## Fixed issues
Fixes #5600

### CODE REVIEW

**Comment by maxkatz6** [comment]
cc @Gillibald

File: src/Avalonia.Controls/TextBox.cs [hunk 1]
```diff
@@ -1047,11 +1048,21 @@ private void HandleTextInput(string? input)
...
 
-        private string? RemoveInvalidCharacters(string? text)
+        private string? SanitizeInputText(string? text)
         {
             if (text is null)
                 return null;
 
+            if (!AcceptsReturn)
+            {
+                var endOfFirstLine = text.IndexOfAny(crlf);
```
Comment (Gillibald) [inline src/Avalonia.Controls/TextBox.cs:1058]:
```
 var linebreakEnumerator = new LineBreakEnumerator(text.AsSpan());

 while(linebreakEnumerator.MoveNext(out var linebreak))
 {
     if (linebreak.Required)
     {
         text = text.Substring(0, Math.Max(0, linebreak.PositionMeasure - 1));

         break;
     }
 }
```

Comment (Gillibald) [reply src/Avalonia.Controls/TextBox.cs:1058]: This is the way you should look for line breaks

**Review by Gillibald — APPROVED** [review]

**Comment by markl65536** [comment]
The tests fail on Windows because `LineBreakEnumerator` does not handle the line breaks as expected on Windows (I develop on Linux and tests pass fine there). Windows uses CRLF as line terminator but `LineBreakEnumerator` leaves a trailing CR. Referring to [Unicode Documentation](https://www.unicode.org/reports/tr14/tr14-37.html) this may be according to spec, because a CR does not cause a line break when followed by a LF, but it is probably not what we want for `TextBox`.

I am at a loss here. Should I add additional code for "cleanup" of extraneous trailing CRs? This is kinda ugly but I do not see a better way when using `LineBreakEnumerator`. Any suggestions?

(Notice that in deviation to the suggestion of @Gillibald I am using `PositionWrap` instead of `PositionMeasure`. `PositionMeasure` would trim all trailing whitespace. That would solve the CR-problem but it would kill trailing space.)

**Comment by Gillibald** [comment]
I have adjusted the logic to only look for line-break characters

**Pull request merged**